app/Importer/AssetModelImporter.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace App\Importer;
+
+use App\Models\AssetModel;
+use App\Models\Depreciation;
+use App\Models\CustomFieldset;
+use Illuminate\Support\Facades\Log;
+
+/**
+ * When we are importing users via an Asset/etc import, we use createOrFetchUser() in
+ * Importer\Importer.php. [ALG]
+ *
+ * Class LocationImporter
+ */
+class AssetModelImporter extends ItemImporter
+{
+    protected $models;
+
+    public function __construct($filename)
+    {
+        parent::__construct($filename);
+    }
+
+    protected function handle($row)
+    {
+        parent::handle($row);
+        $this->createAssetModelIfNotExists($row);
+    }
+
+    /**
+     * Create a model if a duplicate does not exist.
+     * @todo Investigate how this should interact with Importer::createModelIfNotExists
+     *
+     * @author A. Gianotto
+     * @since 6.1.0
+     * @param array $row
+     */
+    public function createAssetModelIfNotExists(array $row)
+    {
+
+        $editingAssetModel = false;
+        $assetModel = AssetModel::where('name', '=', $this->findCsvMatch($row, 'name'))->first();
```

Comment:
Is 'name' alone enough to guarantee uniqueness? I seem to remember us having problems where users had two identically-named models, but with different model numbers, and wanted to maintain those separately? Or I could be misremembering.

Reply:
Hm, yeah, that's a good point...